src/app/lecture/knowledge-node/knowledge-node.component.ts

```diff
@@ -8,11 +11,21 @@ import { KnowledgeNode } from './model/knowledge-node.model';
 })
 export class KnowledgeNodeComponent implements OnInit {
 
-  @Input() knowledgeNode: KnowledgeNode;
+  knowledgeNode: KnowledgeNode;
 
-  constructor() { }
+
+  constructor(
+    public converter: LearningObjectMapper,
```

Comment:
Agreed. The question I have is: Should I keep the mapping logic encapsulated inside a separate class and call it inside the LectureService, or move the mapping logic entirely to LectureService class?

Reply:
We can leave it as is for now. The mapping logic will grow as we add new widgets, and if we decide to use it in multiple places later it will be easier to reuse.